spec/api_v1.yaml

```diff
@@ -0,0 +1,368 @@
+                  error:
+                    type: string
+                    example: The requested resource was not found
+    patch:
+      description: Updates a request (requires special authorization)
+      parameters:
+      - name: id
+        in: path
+        required: true
+        description: The ID of the Cachito request to retrieve
+        schema:
+          type: integer
+      responses:
+        "200":
+          description: The request was updated
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  dependencies:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/Dependency"
+                  environment_variables:
+                    $ref: "#/components/schemas/EnvironmentVariable"
+                  pkg_managers:
+                    type: array
+                    items:
+                      type: string
+                      example: gomod
+                  state:
+                    type: string
+                    example: complete
+                  state_reason:
+                    type: string
+                    example: Completed successfully
+        "404":
+          description: The request wasn't found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The requested resource was not found
+        "500":
+          description: The input is invalid
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The input data must be a JSON object
+      requestBody:
+        description: The keys to update
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RequestNew'
+      security:
+      - "Kerberos Authentication": []
+  "/requests/{id}/download":
+    get:
+      description: Downloads the bundle for the specified Cachito request
+      parameters:
+      - name: id
+        in: path
+        required: true
+        description: The ID of the Cachito request
+        schema:
+          type: integer
+      responses:
+        "200":
+          description: Downloads the bundle
+          content:
+            application/gzip: {}
+        "404":
+          description: The request wasn't found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The requested resource was not found
+        "500":
+          description: The input is invalid
```

Comment:
Are we really returning 500 on user provided input? We should fix that.

Reply:
That was a typo on my part, and then copied to the rest of the document. It's actually a 400 status code. I'll fix it.